### PR TITLE
created a c++ file and updated readme

### PR DESCRIPTION
**Pull request by kalash2205**
## OOPS-constructors and destructors
- #239 

## Proposed Changes
- Created a c++ file with class containing constructors and destructors
- Updated readme with necessary documentation

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the documentation.
- [x] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- Created a c++ file with class containing constructors viz default, parameterized, copy constructors and destructors
- Updated readme with necessary documentation like what are constructors and its types and destructors

- I used example in main function test the constructors

- I ran on vs code

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| CODE  | OUTPUT    |
| 
image
 | 
image
  |

### CODE REVIEW

**Review by Lakhankumawat — APPROVED** [review]
LGTM Mentors please Review the code and PR

File: O-OOPS/README.md [hunk 1]
```diff
@@ -1 +1,36 @@
...
+    - [Parameterized Constructors](#parameterized-constructors)
+    - [Copy Constructors](#copy-constructors)
+- [DESTRUCTORS](#destructors)
+
+
+# CONSTRUCTORS
+
+- Constructors are used to initialise the objects of a class. 
+- It has the same name as the class name.
+- A default constructor is generated by the compiler every time the object of a class is created
```
Comment (n4i9kita) [inline O-OOPS/README.md:14]: I think this line is redundant as you have mentioned in line no 19 as well.

File: O-OOPS/README.md [hunk 2]
```diff
@@ -1 +1,36 @@
...
+
+### Parameterized Constructors
+
+- A contructor in which we can pass arguments to initialise the object. - A parameterized contructor is created just like a normal function.
+
+
+### Copy Contructors
+
+- A copy constructor is simply, creating a copy of an object of a class or initialising an object using another object of the same class.
+- It is used to pass the object by value to any function. BUT the copy contructor function takes its argument by reference because every time a copy constructor is called it will call a copy constructor and this will go so on without terminating.
```
Comment (n4i9kita) [inline O-OOPS/README.md:30]: I think you need to add the word `otherwise` between 'because' and 'every'.

File: O-OOPS/README.md [hunk 3]
```diff
@@ -1 +1,36 @@
...
+
+
+# DESTRUCTORS
+
+- It is a function to destroy the object created. It is called automatically when the program ends or function ends.
```
Comment (n4i9kita) [inline O-OOPS/README.md:35]: I think we can say that ` a destructor is called when an object goes out of scope or is explicitly deleted`.
What do you think?

**Comment by n4i9kita** [comment]
The code is written really nicely. 
Good job.

**Comment by kalash2205** [comment]
done the changes @n4i9kita, @Poojaamangal @miraj0507, @Lakhankumawat

**Review by miraj0507 — COMMENTED** [review]
All checks passes, LGTM

**Review by Lakhankumawat — APPROVED** [review]
LGTM 🌟💯 

Great work @kalash2205 

Although you have pushed some extra files like exe and settings.json but no issues i will delete them but keep in mind to remove these files in next PR .

**Comment by Lakhankumawat** [comment]
@all-contributors please add @kalash2205  code ,

**Pull request merged**